proxy/src/outbound.rs

```diff
@@ -38,33 +41,52 @@ impl<B> Outbound<B> {
     }
 }
 
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub enum Destination {
+    LocalSvc(FullyQualifiedAuthority),
+    External(SocketAddr),
+}
+
 impl<B> Recognize for Outbound<B>
 where
     B: tower_h2::Body + 'static,
 {
     type Request = http::Request<B>;
     type Response = bind::HttpResponse;
     type Error = <Self::Service as tower::Service>::Error;
-    type Key = (FullyQualifiedAuthority, Protocol);
+    type Key = (Destination, Protocol);
     type RouteError = ();
     type Service = Buffer<Balance<
         load::WithPendingRequests<Discovery<B>>,
         choose::PowerOfTwoChoices<rand::ThreadRng>,
     >>;
 
     fn recognize(&self, req: &Self::Request) -> Option<Self::Key> {
-        req.uri().authority_part().map(|authority| {
-            let auth = FullyQualifiedAuthority::new(
+        let local = req.uri().authority_part().and_then(|authority| {
+            FullyQualifiedAuthority::new(
                 authority,
                 self.default_namespace.as_ref().map(|s| s.as_ref()),
-                self.default_zone.as_ref().map(|s| s.as_ref()));
-
-            let proto = match req.version() {
-                http::Version::HTTP_2 => Protocol::Http2,
-                _ => Protocol::Http1,
-            };
-            (auth, proto)
-        })
+                self.default_zone.as_ref().map(|s| s.as_ref()))
+
+        });
+
+        let dest = if let Some(local) = local {
+            Destination::LocalSvc(local)
+        } else {
+            let orig_dst = req.extensions()
```

Comment:
Why do we look up `orig_dst_if_not_local()` here instead of in `bind_service()` below. Regardless of where we do it, let's add a comment explaining why we do it where we do it.